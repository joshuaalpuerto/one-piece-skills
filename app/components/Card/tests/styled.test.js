/**
 * Testing our Modal component
 */

import React from 'react'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'

import {
  DeleteButton,
  IdWrapper,
  ContentWrapper,
  Remove,
  Wrapper
} from '../styled'

const children = 'Test'
const wrapper = (Component, props = {}, enzyme = shallow) =>
  enzyme(<Component {...props}>{children}</Component>)

describe('<Card /> Styled', () => {
  describe('<Wrapper />', () => {
    it('render without exploding', () => {
      const renderComponent = wrapper(Wrapper, {})
      expect(renderComponent.length).toEqual(1)
    })
  })

  describe('<Remove />', () => {
    it('render without exploding', () => {
      const renderComponent = wrapper(Remove, {})
      expect(renderComponent.length).toEqual(1)
    })
  })

  describe('<ContentWrapper />', () => {
    it('render without exploding', () => {
      const renderComponent = wrapper(ContentWrapper, {})
      expect(renderComponent.length).toEqual(1)
    })
  })

  describe('<IdWrapper />', () => {
    it('render without exploding', () => {
      const renderComponent = wrapper(IdWrapper, {})
      expect(renderComponent.length).toEqual(1)
    })

    it('should have display color passed', () => {
      const renderComponent = wrapper(IdWrapper, { bgColor: '#000' }, mount)
      expect(renderComponent).toMatchSnapshot()
      expect(renderComponent).toHaveStyleRule('background-color', '#000')
    })
  })

  describe('<DeleteButton />', () => {
    it('render without exploding', () => {
      const renderComponent = wrapper(DeleteButton, {})
      expect(renderComponent.length).toEqual(1)
    })
  })
})
