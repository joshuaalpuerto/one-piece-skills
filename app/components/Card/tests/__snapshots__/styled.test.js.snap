// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Styled <IdWrapper /> should have display color passed 1`] = `
.c0 {
  background-color: #000;
  color: #fff;
  padding: 5px;
}

<styled__IdWrapper
  bgColor="#000"
>
  <StyledComponent
    bgColor="#000"
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "styled__IdWrapper-dngt2t-1",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "background-color:",
            [Function],
            ";color:#fff;padding:5px;",
          ],
        },
        "displayName": "styled__IdWrapper",
        "render": [Function],
        "styledComponentId": "styled__IdWrapper-dngt2t-1",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      Test
    </div>
  </StyledComponent>
</styled__IdWrapper>
`;
